packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts

```diff
@@ -137,22 +137,34 @@ export class GutenbergEditorPage {
 	}
 
 	/**
-	 * Return the editor iframe.
+	 * Return the editor frame. Could be the top-level frame.
+	 * an iframe (Gutenframe).
 	 *
-	 * @returns {Promise<Frame>} iframe holding the editor.
+	 * @returns {Promise<Frame>} frame holding the editor.
 	 */
 	async getEditorFrame(): Promise< Frame > {
-		const locator = this.page.locator( selectors.editorFrame );
-
-		const elementHandle = await locator.elementHandle( {
-			timeout: 105 * 1000,
-		} );
-
-		if ( ! elementHandle ) {
-			throw new Error( 'Could not locate editor iframe.' );
+		// If we're not in the Gutenframe context (i.e not Calypso), then
+		// just return the top frame, no need to try to locate the iframe.
+		if ( ! this.page.url().startsWith( 'https://wordpress.com' ) ) {
+			return await this.page.mainFrame();
 		}
 
-		return ( await elementHandle.contentFrame() ) as Frame;
+		const locator = this.page.locator( selectors.editorFrame );
+
+		return await locator
+			.elementHandle( {
+				timeout: 105 * 1000,
+			} )
+			.then( async ( elementHandle ) => ( await elementHandle!.contentFrame() ) as Frame )
+			.catch( async () => {
+				// The CalipsoifyIframe component will redirect to the plain WPAdmin page if
+				// there's an error or if the iframe takes too long to completely load. The
+				// goal here is to ge hold of the editor and not test any aspects related to
+				// the Gutenframe, so we return the `mainFrame` to allow the test to access
+				// the editor in the WPAdmin page.
+				console.info( 'Could not locate editor iframe. Returning the top-level frame instead' );
+				return await this.page.mainFrame();
```

Comment:
What led me to try this was the fact that my AT test site was redirecting to WPadmin after a few seconds. The test would then fail after the timeout because the Gutenframe couldn't be found. This works well as a fallback for when Calypso ends up redirecting to WP-Admin. It will return the `mainFrame()` where the Gutenberg editor can be found and everything else will work accordingly.

I'm not 100% sure about this yet, though. I can see a few potential issues:
1. One is of test intent - one could argue that abstracting this logic is not good because we should know when the iframe is not loading and the test should fail. However, you could also argue that this component is interested in the editor and not how to get hold of it, and the actual testing of the iframe logic should be done in a separate test/test component;
2. Is of performance. The fallback only takes place after 105000ms (when the timeout exceptions are finally thrown). That can add up to the test time. Could we possibly reduce the timeout?

It might as well be better to assume that simple and AT (test) sites will always load through Calypso (and hence, the Gutenframe) and if it's redirected to WPAdmin, then something is wrong. Then, we could have a specific suite of tests that test Gutenframe itself (though it might be better to look into adding it as further down the test pyramid, at first glance, I can't find any tests for the https://github.com/Automattic/wp-calypso/blob/trunk/client/gutenberg/editor/calypsoify-iframe.tsx nor https://github.com/Automattic/wp-calypso/blob/trunk/apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js. In the end, it might just be overkill to work on this now, though it's worth discussing.

@Automattic/quality-squad looking forward to reading your thoughts about this.

Reply:
@dpasque Thanks for the reply! 

> Can we have the best of both worlds? Could we expose an option in the constructor of the GutenbergEditorPage that controls whether this throws or not? That way, tests that need the resiliency could have it, but we also have an easy way to build out specs that specifically look for the redirect away from the Gutenframe. This feels like an appropriate use of constructor options to me personally!

Sounds good! From what I understood, we'd have this resiliency logic activate by default, right?

Reply:
@dpasque Done 79738e23f5aff43c0835012d9f068a4431bc9b46. Let me know what you think :)

Reply:
>we'd have this resiliency logic activated by default, right?

Yes, that's what I imagined!

>@dpasque Done https://github.com/Automattic/wp-calypso/commit/79738e23f5aff43c0835012d9f068a4431bc9b46. Let me know what you think :)

I think this looks really nice!! 🎉  Thanks for adding that! 😄 

My one suggestion is maybe we could call it `throwIfNotGutenframe` because there may be cases where the `/wp-admin` route is loaded directly? As in, someone may not actually get the boot! 🥾

Reply:
> My one suggestion is maybe we could call it throwIfNotGutenframe because there may be cases where the /wp-admin > route is loaded directly? As in, someone may not actually get the boot! hiking_boot

Ah, that makes sense!

Reply:
I followed your suggestion in principle but called the param/property `requiresGutenframe`. I personally don't like naming things with a negation, as it makes it can make actually negating them in the code a bit confusing to reason about. I've done it in https://github.com/Automattic/wp-calypso/pull/61349/commits/a688ea19057334b5e4878efff668d87375280cd9 (check the commit description also). Let me know your thoughts about!

Reply:
Ooo, that reads so nicely!! Love it! 🎉